sound/soc/sof/topology.c

```diff
@@ -751,14 +751,15 @@ static const struct sof_topology_token led_tokens[] = {
 	 get_token_u32, offsetof(struct snd_sof_led_control, direction), 0},
 };
 
-static void sof_parse_uuid_tokens(struct snd_soc_component *scomp,
-				  void *object,
-				  const struct sof_topology_token *tokens,
-				  int count,
-				  struct snd_soc_tplg_vendor_array *array)
+static int sof_parse_uuid_tokens(struct snd_soc_component *scomp,
```

Comment:
comment on commit message: I have absolutely no idea if you describe a data corruption issue ("copied on top"), a bad list parsing (returns last instead of first).

What are 'multiple similar tuple arrays'? why is this useful?
What are you trying to fix or enable really?

Reply:
Yeah, sorry for being too sparse. I tried to explain this above also... so I don't think currently the parser handles multiple similar tuples arrays following each other in topology. Every array is parsed but copied on top of the previous one. And you can't do that easily as parser doesn't tell that it has found something or the position of the found array. So our parser will always run to the end of given data section despite it found 1,2,3,4 instances of the array and the last one of those is what you get. AFAIK currently there is just no way of parsing the array number 1,2,3 as the parser will run to the end of data.

As said we've not had this kind of situation, because our components don't currently have this kind of multiple array structure. Mux/demux is the first one with this kind of more complex structure.